XLS-0081d-permissioned-dex/README.md

```diff
@@ -0,0 +1,593 @@
+
+### 1.4. How Domains Work
+
+A permissioned offer will only be filled by valid domain offers.
+
+An open offer can be filled by any open offer on the open DEX, but cannot be filled by a permissioned offer. Likewise, any permissioned offer cannot be filled by an open offer.
+
+## 2. On-Ledger Object: `Offer`
+
+The `Offer` object tracks an offer placed on the CLOB DEX. This object type already exists on the XRPL, but is being extended as a part of this spec to also support permissioned DEX domains.
+
+### 2.1. Fields
+
+<details>
+<summary>
+
+As a reference, [here](https://xrpl.org/docs/references/protocol/ledger-data/ledger-entry-types/offer) are the existing fields for the `Offer` object.
+</summary>
+
+| Field Name | Required? | JSON Type | Internal Type | Description |
+|------------|-----------|-----------|---------------|-------------|
+|`Account`|✔️|`string`|`AccountID`|The address of the account that owns this offer.|
+|`BookDirectory`|✔️|`string`|`Hash256`|The ID of the offer directory that links to this offer.|
+|`BookNode`|✔️|`string`|`UInt64`|A hint indicating which page of the offer directory links to this entry, in case the directory consists of multiple pages.|
+|`Expiration`| |`number`|`UInt32`|Indicates the time after which this offer is considered unfunded.|
+|`LedgerEntryType`|✔️|`string`|`UInt16`|The value `0x006F`, mapped to the string `Offer`, indicates that this is an offer entry.|
+|`OwnerNode`|✔️|`string`|`UInt64`|A hint indicating which page of the owner directory links to this entry, in case the directory consists of multiple pages.|
+|`PreviousTxnID`|✔️|`string`|`Hash256`|The identifying hash of the transaction that most recently modified this entry.|
+|`PreviousTxnLgrSeq`|✔️|`number`|`UInt32`|The ledger index that contains the transaction that most recently modified this object.|
+|`Sequence`|✔️|`number`|`UInt32`|The `Sequence` value of the `OfferCreate` transaction that created this offer. Used in combination with the `Account` to identify this offer.|
+|`TakerPays`|✔️|`object`|`Amount`|The remaining amount and type of currency requested by the offer creator.|
+|`TakerGets`|✔️|`object`|`Amount`|The remaining amount and type of currency being provided by the offer creator.|
+</details>
+
+This spec proposes deprecating the `BookDirectory` and `BookNode` fields and adding the following:
+
+| Field Name | Required? | JSON Type | Internal Type | Description |
+|------------|-----------|-----------|---------------|-------------|
+|`DomainID`| |`string`|`Hash256`|The domain that the offer must be a part of.|
+|`BookDirectories`|✔️|`string`|`Vector256`|The IDs of the offer directories that link to this offer.|
+|`BookNodes`|✔️|`string`|`STArray`|Hints indicating which pages of the offer directory links to this entry, in case the directories consist of multiple pages.|
```

Comment:
what are the use cases of this? when is an offer included in multiple directories?

Reply:
it's going to a lot of dirty work to deprecate `BookDirectory`, not sure if the effort is worth it if there's not strong reasoning for this